tests/unit/command/test_update.py

```diff
@@ -1,15 +1,35 @@
+import logging
+
 from dvc.cli import parse_args
 from dvc.command.update import CmdUpdate
+from dvc.stage import UpdateWithRevNotPossibleError
 
 
 def test_update(dvc_repo, mocker):
     targets = ["target1", "target2", "target3"]
-    cli_args = parse_args(["update"] + targets)
+    cli_args = parse_args(["update", "--rev", "develop"] + targets)
     assert cli_args.func == CmdUpdate
     cmd = cli_args.func(cli_args)
     m = mocker.patch("dvc.repo.Repo.update")
 
     assert cmd.run() == 0
 
-    calls = [mocker.call(target) for target in targets]
+    calls = [mocker.call(target, rev="develop") for target in targets]
     m.assert_has_calls(calls)
+
+
+def test_update_rev_failed(mocker, caplog):
+    targets = ["target1", "target2", "target3"]
+    cli_args = parse_args(["update", "--rev", "develop"] + targets)
+    assert cli_args.func == CmdUpdate
+
+    cmd = cli_args.func(cli_args)
+    with mocker.patch.object(
+        cmd.repo, "update", side_effect=UpdateWithRevNotPossibleError()
+    ):
+        with caplog.at_level(logging.ERROR, logger="dvc"):
+            assert cmd.run() == 1
+            expected_error = (
+                "Revision option (--rev) is not a feature of non-Git sources."
```

Comment:
```suggestion
                "Revision option (`--rev`) is not a feature of non-Git source projects."
```

Reply:
@ilgooz please note my suggestion is pretty cosmetic/ not blocked by the discussion in the issue. If you accept this suggestion though, please remember to update the cmd ref (in docs PR) as well. Thanks!

Reply:
Thanks Jorge, sure thing! I did notice that I forgot this one but already fixed in my local. I'll push the changes in a bit.

Reply:
done by 32494bb

Reply:
done by a9fa7975

Reply:
Np. And thanks but I don't see any changes to this string yet. And I guess it would also need to be updated where it's originally printed? Not just in the test.

Reply:
Ah, I see the changes now.

Reply:
Yeah, I forgot it to git add in the first try, haha. 👨‍🎨